src/resmom/mock_run.c

```diff
@@ -235,3 +237,41 @@ mock_run_mom_set_use(job *pjob)
 
 	return (PBSE_NONE);
 }
+
+/**
+ * @brief	job_purge for mock run mode
+ *
+ * @param[in,out]	pjob - the job being purged
+ *
+ * @return	void
+ */
+void
+mock_run_job_purge(job *pjob)
+{
+	char namebuf[MAXPATHLEN + 1] = {'\0'};
+
+	delete_link(&pjob->ji_jobque);
+	delete_link(&pjob->ji_alljobs);
+	delete_link(&pjob->ji_unlicjobs);
+
+#ifdef PBS_MOM
```

Comment:
why is there an ifdef for MOM, is this function called in server too?

Reply:
'ji_preq' field of pjob is wrapped around #ifdef PBS_MOM, so if I don't wrap it here then compiler barfs